raiden/storage/sqlite.py

```diff
@@ -876,9 +877,15 @@ def get_events(self, limit: int = None, offset: int = None) -> List[Event]:
         events = self.database.get_events(limit, offset)
         return [self.serializer.deserialize(event) for event in events]
 
-    def get_state_changes(self, limit: int = None, offset: int = None) -> List[StateChange]:
-        state_changes = self.database.get_state_changes(limit, offset)
-        return [self.serializer.deserialize(state_change) for state_change in state_changes]
+    def get_state_changes_stream(
+        self, retry_timeout: float, limit: int = None, offset: int = 0
+    ) -> Iterator[List[StateChange]]:
+        while True:
+            state_changes = self.database.get_state_changes(limit, offset)
```

Comment:
I checked and this does a `SELECT ... ORDER BY` on an indexed column, so it should be quite efficient.